app/src/utils/hooks.ts

```diff
@@ -138,6 +138,40 @@ export const usePageParams = () => {
   return { page, pageSize, setPageParams };
 };
 
+export const useVisibleEvaluationColumns = () => {
+  const router = useRouter();
+
+  // Split the "compare" query by commas to get the array of strings.
+  const visibleColumns =
+    typeof router.query.compare === "string"
+      ? router.query.compare.split(",").filter((col) => !!col)
```

Comment:
I suspect at this level of complexity we'd be better served by adopting something like https://www.npmjs.com/package/use-query-params but haven't actually used it so I guess I'm not sure it'll make the code simpler/do exactly what we want.

Reply:
We were actually using that package a few weeks ago, but I found that it had an odd interaction with NextJS that caused the [id] portion of a route to be shown as a query param in addition to being shown in the path when we used non-replacing query param updates.